py2k/creators.py

```diff
@@ -91,15 +91,22 @@ def __init__(self,
         self._base = base
 
     def create(self):
-        records_as_dict_list = self._df.to_dict('records')
-        if records_as_dict_list:
-            sample_record = records_as_dict_list[0]
-            model = self._create_model(sample_record)
-            return model
-        else:
+        if self._df.empty:
             raise ValueError(
                 "Unable to create kafka model from an empty dataframe.")
 
+        sample_record = {
+            col: self._sample_col(col) for col in self._df.columns
+        }
+        return self._create_model(sample_record)
+
+    def _sample_col(self, col):
+        without_na = self._df[col].dropna()
```

Comment:
Wouldn't it be a bit inefficient if the dataframe is to big, i.e. isn't `dropna` going through the whole column? Maybe iterating here would be more efficient?

Reply:
Good question, and I did a quick analysis before I used it:

![obrazek](https://user-images.githubusercontent.com/8339148/125084327-abc82100-e0c9-11eb-94a7-b7e56fee3bd1.png)


As you can see, the iteration is faster when there is less than 1000 Nones before the first Non-None.
The reasoning behind choosing dropna is: When there is fewer Nones, the execution time is already "fast enough".

But I get your point that when it is more likely that there will be fewer null values at the beginning, the iteration is way faster.

Happy to change if you guys think the iteration makes more sense. @DanWertheimer what do you think?